rs/nervous_system/tools/sync-with-released-nervous-system-wasms/src/main.rs

```diff
@@ -79,68 +100,100 @@ async fn get_mainnet_canister_git_commit_id_and_module_hash(
             .read_state_canister_info(canister_id, "module_hash")
             .await?
     };
-    let module_hash_str = module_hash.iter().fold(String::new(), |mut output, x| {
-        let _ = write!(output, "{:02x}", x);
-        output
-    });
 
-    Ok((git_commit_id, module_hash_str))
+    Ok((git_commit_id, module_hash_hex(module_hash)))
 }
 
-/// This function fetches the given canister's module hash deployed on the ICP mainnet (based on `canister_id`)
-/// and then finds a git tag in the given `canister_repository` (e.g., `dfinity/cycles-ledger`)
-/// which contains a release asset whose name matches `canister_name`
-/// and whose sha256 hash matches the module hash deployed on the ICP mainnet
-/// (the sha256 hash is fetched from a dedicated release asset whose name has the form `{canister_name}.sha256`).
-/// To find the git tag, the function proceeds as follows:
-///   - it crawls all git tags of the given `canister_repository`;
-///   - for every git tag, it lists all release assets and looks for a release asset whose name has the form `{canister_name}.sha256`:
-///     if that release asset contains the expected module hash (as deployed on the ICP mainnet),
-///     then the corresponding tag is returned.
-async fn get_mainnet_canister_git_tag_and_module_hash(
-    agent: &Agent,
-    canister_id: CanisterId,
-    canister_repository: String,
-    canister_name: String,
-) -> Result<(String, String)> {
-    use std::fmt::Write;
+const GITHUB_API: &str = "https://api.github.com";
 
-    #[derive(Debug, Deserialize)]
-    struct Tag {
-        name: String,
-    }
+fn github_api_client_and_token() -> Result<(Client, String)> {
+    // GitHub API requires all requests to provide a user-agent header
+    // so we fix an arbitrary value of this header here.
+    let client = Client::builder()
+        .user_agent("sync-with-released-nervous-system-wasms")
+        .build()?;
 
-    #[derive(Debug, Deserialize)]
-    struct ReleaseAsset {
-        name: String,
-        browser_download_url: String,
-    }
+    let token = std::env::var("GITHUB_TOKEN").expect("Set GITHUB_TOKEN env var");
+
+    Ok((client, token))
+}
+
+// GitHub API types
+
+#[derive(Debug, Deserialize)]
+struct Tag {
+    name: String,
+}
+
+#[derive(Clone, Debug, Deserialize)]
+struct ReleaseAsset {
+    name: String,
+    browser_download_url: String,
+}
 
-    #[derive(Debug, Deserialize)]
-    struct Release {
-        tag_name: String,
-        assets: Vec<ReleaseAsset>,
+impl ReleaseAsset {
+    // Returns the sha256 hash of the asset content.
+    async fn sha256(&self) -> Result<String> {
+        let (client, token) = github_api_client_and_token()?;
+
+        let asset_bytes = client
```

Comment:
Nit. We could make this function cache the result, since it can be expensive to recompute multiple times

Reply:
How could we do that?

Reply:
For the current use case, caching isn't needed. But it's possible that now that the code is nice, people will want to reuse it. I don't request any changes, just mentioning this as I was thinking about reusability while reviewing the PR.